backend/hdf5/FileHDF5.cpp

```diff
@@ -333,30 +333,45 @@ shared_ptr<base::IFile> FileHDF5::file() const {
 }
 
 
-bool FileHDF5::checkHeader(FileMode mode) {
+bool FileHDF5::checkHeader(FileMode mode, std::string &message) {
     bool check = true;
     vector<int> vv;
     string str;
     if (root.hasAttr("format")) {
         if (!root.getAttr("format", str) || str != FILE_FORMAT) {
             check = false;
+            message = "File is not a valid NIX file, format mismatch!";
         }
     } else {
         check = false;
+        message = "File is not a valid NIX file, format attribute missing!";
     }
     if (check && root.hasAttr("version")) {
         if (!root.getAttr("version", vv)) {
             check = false;
+            message = "File is not a valid NIX file, could not read version attribute!";
         } else {
             file_format_version = FormatVersion(vv);
             if (mode == FileMode::ReadWrite) {
                 check = my_version.canWrite(file_format_version);
+                if (!check) {
+                    message = "Cannot open file for ReadWrite access, format mismatch! ";
+                    std::stringstream stream;
```

Comment:
If you have a `stringstream` already, maybe better to use that for constructing the message instead of using `message.append`?